### DISCUSSION TITLE
Difference between Either and Result

### DISCUSSION THREAD

**Discussion by juchom** [louthy/language-ext#1226] [Q&A] [answered]
Hello,

I try to slowly introduce LanguageExt in my project and I'm not sure if Either and Result are the same thing.

When should we use one over the other ?

Thanks

**Comment by louthy** [comment, marked as answer]
Writing this response on my phone, so apologies for brevity and lack of markup/formatting. 

Never use Result is the short answer.  Result may be deprecated at some point (because this question comes up a lot and clearly causes confusion), mostly because it was only ever designed as an intermediate type for Try, not as a fully fledged monadic type (like Either or Fin)

That's also the reason it sits in a different namespace, so it wouldn't be used accidentally; and so it wouldn't cause naming conflicts with other libraries that use Result (very common name).

`Fin<A>` is actually closer to `Result<A>`, it is like `Either<Error, A>`.  It's called Fin to be short and easy to use, unlikely to conflict with other libs, and be consider 'final' like 'result'.

So Fin is the one to use 👍

(I notice you're French, so I hope you appreciate the etymology too 😉)

**Reply by juchom** [reply]
Ok thanks for the quick reply !

What may avoid this question again is to add a README.md here : LanguageExt.Core/Common/Result like you did in other parts of the project. I went here before asking the question.

> (I notice you're French, so I hope you appreciate the etymology too 😉)

It's even clearer now :)

Again, thanks a lot for the quick reply and this great project !
